modules/js/src/loader.js

```diff
@@ -1,108 +1,98 @@
+    }
 
-        if ("simd" in paths) {
-            self.simdPath = paths["simd"];
-        }
+    if ("threadsSimd" in paths) {
+        threadsSimdPath = paths["threadsSimd"];
+    }
 
-        if ("mtSIMD" in paths) {
-            self.mtSIMDPath = paths["mtSIMD"];
-        }
+    let wasmSupported = !(typeof WebAssembly === 'undefined');
+    if (!wasmSupported && OPENCV_URL == "" && asmPath != "") {
+        OPENCV_URL = asmPath;
+        console.log("The OpenCV.js for Asm.js is loaded now");
+    } else if (!wasmSupported && asmPath == ""){
+        throw new Error("The browser supports the Asm.js only, but the path of OpenCV.js for Asm.js is empty");
     }
 
-    this.loadOpenCV = function (onloadCallback) {
-        let OPENCV_URL = "";
-        let wasmSupported = self.judgeWASM();
-        let simdSupported = self.judgeSIMD();
-        let threadsSupported = self.judgeThreads();
+    let simdSupported = wasmSupported ? await wasmFeatureDetect.simd() : false;
+    let threadsSupported = wasmSupported ? await wasmFeatureDetect.threads() : false;
 
-        if (!wasmSupported && OPENCV_URL == "" && self.asmPath != "") {
-            OPENCV_URL = asmPath;
-        } else if (!wasmSupported && self.asmPath == ""){
-            throw new Error("The browser supports the Asm.js only, but the path of OpenCV.js for Asm.js is empty");
+    if (simdSupported && threadsSupported && threadsSimdPath != "") {
+        OPENCV_URL = threadsSimdPath;
+        console.log("The OpenCV.js with simd and threads optimization is loaded now");
+    } else if (simdSupported && simdPath != "") {
+        if (threadsSupported && threadsSimdPath == "") {
+            console.log("The browser supports simd and threads, but the path of OpenCV.js with simd and threads optimization is empty");
         }
-
-        if (simdSupported && threadsSupported && OPENCV_URL == "" && self.mtSIMDPath != "") {
-            OPENCV_URL = self.mtSIMDPath;
-        } else if (simdSupported && threadsSupported && OPENCV_URL == "") {
-            throw new Error("The browser supports simd and threads, but the path of OpenCV.js with simd and threads optimization is empty");
+        OPENCV_URL = simdPath;
+        console.log("The OpenCV.js with simd optimization is loaded now.");
+    } else if (threadsSupported && threadsPath != "") {
+        if (simdSupported && threadsSupported) {
+            console.log("The browser supports simd and threads, but the path of OpenCV.js with simd and threads optimization is empty");
         }
-
-        if (simdSupported && OPENCV_URL == "" && self.simdPath != "") {
-            OPENCV_URL = self.simdPath;
-        } else if (simdSupported && OPENCV_URL == "") {
-            throw new Error("The browser supports simd, but the path of OpenCV.js with simd optimization is empty");
+        OPENCV_URL = threadsPath;
+        console.log("The OpenCV.js with threads optimization is loaded now");
+    } else if (threadsPath) {
```

Comment:
`else if (threadsSupported)`?

Reply:
Sorry, it is my fault. This choice should be deleted.